platforms/linux/aarch64-gnu.toolchain.cmake

```diff
@@ -0,0 +1,83 @@
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_VERSION 1)
+set(CMAKE_SYSTEM_PROCESSOR aarch64)
+
+set(GCC_COMPILER_VERSION "5" CACHE STRING "GCC Compiler version")
+
+find_program(CMAKE_C_COMPILER NAMES aarch64-linux-gnu-gcc-${GCC_COMPILER_VERSION})
+find_program(CMAKE_CXX_COMPILER NAMES aarch64-linux-gnu-g++-${GCC_COMPILER_VERSION})
+set(ARM_LINUX_SYSROOT /usr/aarch64-linux-gnu CACHE PATH "ARM cross compilation system root")
+
+set(CMAKE_CXX_FLAGS           ""                    CACHE STRING "c++ flags")
+set(CMAKE_C_FLAGS             ""                    CACHE STRING "c flags")
+set(CMAKE_SHARED_LINKER_FLAGS ""                    CACHE STRING "shared linker flags")
+set(CMAKE_MODULE_LINKER_FLAGS ""                    CACHE STRING "module linker flags")
+set(CMAKE_EXE_LINKER_FLAGS    ""                    CACHE STRING "executable linker flags")
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
+set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
+
+set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_SHARED_LINKER_FLAGS}")
+set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_MODULE_LINKER_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS    "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_EXE_LINKER_FLAGS}")
+
+#if(USE_NEON)
+#  message(WARNING "You use obsolete variable USE_NEON to enable NEON instruction set. Use -DENABLE_NEON=ON instead." )
+#  set(ENABLE_NEON TRUE)
+#elseif(USE_VFPV3)
+#  message(WARNING "You use obsolete variable USE_VFPV3 to enable VFPV3 instruction set. Use -DENABLE_VFPV3=ON instead." )
+#  set(ENABLE_VFPV3 TRUE)
```

Comment:
I can remove the commented code block. Essentially when compiling png with its neon optimizations (when ENABLE_NEON is set) it was breaking on AARCH64. Once this patch is in I can dig further into cleaning up AARCH64 + ENABLE_NEON.